appuio/secret/templates/secret.yaml

```diff
@@ -1,16 +1,34 @@
-{{- range .Values.secrets }}
+{{- range $name, $value := .Values.secrets }}
 ---
+{{- with $value }}
 apiVersion: v1
 kind: Secret
 type: {{ default "Opaque" .type }}
 metadata:
-  name: {{ .name }}
+  {{- if .nameTemplate }}
+  name: {{ tpl .nameTemplate $ }}
```

Comment:
I was wondering about that naming of that field. It indeed is treated as a template. Where can I find details on what I can do in that template? What variables can I use?

Reply:
Helm has documentation of how templating works: https://helm.sh/docs/chart_template_guide/functions_and_pipelines/
And more extensive list of functions is here: https://helm.sh/docs/chart_template_guide/function_list/.

Obviously certain functions wouldn't make sense to be used, e.g. there's no point to provide a YAML object to a name with `toYaml`, since that would produce invalid spec for the Secret.

I'm assuming that the average Helm user is aware of these functions, but maybe it's worth adding the link.